### DISCUSSION TITLE
Logging issue Part II

### DISCUSSION THREAD

**Discussion by modezz** [wiggal/GEN24_Ladesteuerung#52] [Q&A]
Servus Wiggal,
beim Logging geht hier der Wert "InBatterie" nicht mit, bei MouseOver wird hier 0 angezeigt. Logging funktioniert:
Es sind 2 Symos und ein Gen24 im einsatz an dem die BYD hängt(19,3KW) - daher die produktion des GEN24 nur 3040 und Einspeisung aber 9395 und 7912 gehen in die BAtterie - Bringt das die Berechnung durcheinander da hier nur vom GEN24 ausgegangen wird? - DANKE

************* BEGINN:  2024-01-27 10:10:03.312735 ************* 

######### L A D E S T E U E R U N G #########

aktuellePrognose:            9934
RestTagesPrognose:           55583
PrognoseAbzugswert/Stunde:   3300
TagesPrognose - Abzugswerte: 13910
Grundlast_Summe für Tag:     3833
aktuellePVProduktion/Watt:   3040
aktuelleEinspeisung/Watt:    9395
aktuelleBatteriePower/Watt:  -7912
GesamtverbrauchHaus/Watt:    -14267
aktuelleBattKapazität/Watt:  13973
Batteriestatus in Prozent:   27.8 %
LadewertGrund:  BattStatusProz < MindBattLad
Bisheriger Ladewert/Watt:    7999
Bisheriger Ladewert/Prozent: 44.64 %
Neuer Ladewert/Watt:         8000
Neuer Ladewert/Prozent:      44.64 %
newPercent_schreiben:        0

Alte und Neue Werte unterscheiden sich weniger als die Schreibgrenzen des WR, NICHTS zu schreiben!!

Daten wurden in die SQLite-Datei gespeichert!
************* ENDE:  2024-01-27 10:10:03.506713 ************* 

![log2](https://github.com/wiggal/GEN24_Ladesteuerung/assets/38989365/078a85f7-ce20-4e33-9dd1-cf10c7954eae)
![logging](https://github.com/wiggal/GEN24_Ladesteuerung/assets/38989365/144f58e3-3265-439b-a189-6e0f1267bf0b)

**Comment by wiggal** [comment]
An der Ecke bin ich aktuell am Basteln, gib mir etwas Zeit. Das Logging müsste passen, die Aufbereitung hackt noch etwas.
wie sieht den die Grafik mit Quelle und Ziel 
Schönen Gruß

**Comment by modezz** [comment]
QZ sieht besser aus aber die Summen verstehe ich entweder nicht oder diese passen noch nicht... 
![qz](https://github.com/wiggal/GEN24_Ladesteuerung/assets/38989365/95e0defb-01d8-4dab-958f-e88f38b10aae)

**Comment by wiggal** [comment]
servus modezz,
ich habs mir jetzt mal kurz angesehen, das Problem hängt mit  den beiden Symos zusammen.
Ich speichere die Daten fürs Logging von der API IP_GEN24:/components/readable von folgenden Werten:

```
In Batterie gespeichert = 393216,CACHE,BAT_ENERGYACTIVE_ACTIVECHARGE_SUM_01_U64
Aus Batterie verbraucht = 393216,CACHE,BAT_ENERGYACTIVE_ACTIVEDISCHARGE_SUM_01_U64
Einspeisung = 16711680,PowerMeter,SMARTMETER_ENERGYACTIVE_PRODUCED_SUM_F64
Netzverbrauch = 16711680,PowerMeter,SMARTMETER_ENERGYACTIVE_CONSUMED_SUM_F64
AC Produktion WR = 327680,Site,ACBRIDGE_ENERGYACTIVE_PRODUCED_SUM_U64
DC Produktion WR = 393216,CACHE,PV_ENERGYACTIVE_ACTIVE_SUM_01_U64 +02_U64
```

Hier kommen die Daten der Produktion nur vom GEN24 und die anderen Daten wie Einspeisung und "In Batterie gespeichert" kommen von der ganzen Anlage. Dadurch ergeben sich in der Berechnung der Verbräuche falsche Werte (z.B. -47,9 KWh von deinem letzten Foto).
Das wirkt sich natürlich, auch auf die Grafiken aus.

Ich habe hierfür aktuell keine Lösung.
Einzige Abhilfe wäre, wenn in der API irgenwo die Werte für die "AC Produktion" bzw. "DC Produktion" für die gesamte Anlage drin wären.
Dies kann ich aber nicht überprüfen, da ich nur einen WR habe und deshalb die Werte immer gleich wären.

Schönen Gruß

**Reply by modezz** [reply]
Servus Wiggal,
es muss irgendwo die variable gesamte AC Produktion geben denn die Limitierung der gesamten Anlage erfolgt über den GEN24, also muss dieser auch die gesamte AC produktion sehen, dieser hat ja auch das Smartmeter per RS485 verbunden.

Muss mir das xml mal abholen, RPI mit teamviewer ist leider eine sehr laggy Lösung(gerade wieder abgeschmiert), sobald ichs transferiert habe poste ich das hier.

DANKE!
SG

**Reply by modezz** [reply]
AC Produktion overall sprich gesamte Anlage:
"SMARTMETER_POWERACTIVE_MEAN_SUM_F64" ist der Aktuell wert des SM ob geliefert(negatives vorzeichen) oder bezogen wird

Summe Bezogen vom Energieversorger (kWh):
Befehlserkennung: \i"SMARTMETER_ENERGYACTIVE_CONSUMED_SUM_F64" : \i\v

Summe Geliefert an den Energieversorger (kWh):
\i"SMARTMETER_ENERGYACTIVE_PRODUCED_SUM_F64" : \i\v

Aktuelle Leistung des Energieversorgers (kW):
\i"SMARTMETER_POWERACTIVE_MEAN_SUM_F64" : \i\v

oder mit
http://ip-wechselrichter/solar_api/v1/GetMeterRealtimeData.cgi?Scope=Device&DeviceId=0
![power](https://github.com/wiggal/GEN24_Ladesteuerung/assets/38989365/6c2ab377-2968-46d9-81fd-b59a52f7cda2)

**Reply by wiggal** [reply]
ja schon, die Werte Netzbezug oder Einspeisung sind ja nicht das Problem.
Damit man den Verbrauch des Hauses berechnen kann braucht man den Wert der Produktion der gesamten PV Anlage, also von allen Wechselrichtern.

**Reply by modezz** [reply]
Ok dann wirds schwierig. Maximal als seperate Anfrage( API) an einen Symo der ja sieht was die 2 Symos Produzieren(Solar.Ring).

**Reply by iReckyy** [reply]
Gibt es hierzu Updates?
Also wen man mehrere Symos verbaut hat, bzw. die Kombi aus Gen24 und Symo?

Ab Version 21.1 kann man ja mehrere Wechselrichter angeben, aber funktioniert das auch mit normalen Symos?

**Reply by wiggal** [reply]
Ob das mit den normalen Symos, funktioniert kann ich nicht sagen, müsste man probieren (ich habe keinen),
oder du kannst mir die API-Datei schicken:
IP_Symo/components/readable

**Reply by iReckyy** [reply]
Bisher läuft bei uns auch nur ein Gen24, die Erweiterung mit einem Symo 4.5-3-M ist aber noch diesen Sommer geplant.

**Reply by wiggal** [reply]
wenn er in Betrieb ist, kannst du ja mal was schicken. Wenn er die nötigen Daten liefert, kann ich es ja auf Symo erweitern, wenns noch nicht passt 😉

**Reply by iReckyy** [reply]
Werde ich tun. 
Bin begeistert von diesem Programm/Support und möchte auf jeden Fall unterstützen.

**Comment by modezz** [comment]
Also, man könnte bei Symos die per DaisyChain verbunden sind wie folgt abfragen:

DeviceId=1 oder 2 je nach WR_ID

http://0.0.0.0/solar_api/v1/GetInverterRealtimeData.cgi?Scope=Device&DeviceId=2&DataCollection=CommonInverterData
Antwortet mit:
{
   "Body" : {
      "Data" : {
         "DAY_ENERGY" : {
            "Unit" : "Wh",
            "Value" : 37780
         },
         "DeviceStatus" : {
            "ErrorCode" : 0,
            "LEDColor" : 2,
            "LEDState" : 0,
            "MgmtTimerRemainingTime" : -1,
            "StateToReset" : false,
            "StatusCode" : 7
         },
         "FAC" : {
            "Unit" : "Hz",
            "Value" : 49.979999999999997
         },
         "IAC" : {
            "Unit" : "A",
            "Value" : 31.5
         },
         "IDC" : {
            "Unit" : "A",
            "Value" : 15.26
         },
         "PAC" : {
            "Unit" : "W",
            "Value" : 7536
         },
         "TOTAL_ENERGY" : {
            "Unit" : "Wh",
            "Value" : 826189
         },
         "UAC" : {
            "Unit" : "V",
            "Value" : 239.59999999999999
         },
         "UDC" : {
            "Unit" : "V",
            "Value" : 639.60000000000002
         },
         "YEAR_ENERGY" : {
            "Unit" : "Wh",
            "Value" : 390000.81
         }
      }
   },
   "Head" : {
      "RequestArguments" : {
         "DataCollection" : "CommonInverterData",
         "DeviceClass" : "Inverter",
         "DeviceId" : "2",
         "Scope" : "Device"
      },
      "Status" : {
         "Code" : 0,
         "Reason" : "",
         "UserMessage" : ""
      },
      "Timestamp" : "2024-01-29T13:33:32+01:00"
   }
}

IDC & UDC parsen, ausmultiplizieren und dazuaddieren - voila?

**Reply by wiggal** [reply]
Geht leider nicht so einfach,

da ich ja keine peroduzierten Werte speichere sondern nur Zählerstände.
Aus den Zählerständen kann man dann jederzeit für alle Zeiträume Werte berechnen.
Hat auch den Vorteil, dass nichts verloren gehen kann, wenn mal aus irgen welchen Gründen die Speicherung der Werte nicht klappt.

Schönen Gruß-

**Reply by modezz** [reply]
Schade,

Hilft es etwas hinsichtlich Ladeberechnung oder schau diese nur auf die Einspeisung?

SG

**Reply by wiggal** [reply]
die Ladeberechung rechnet nur mit der Prognose und dem Batteriestand. Zusätzlich schaut sie drauf, dass die Einspeisung nicht über die Einspeisebegrenzung geht
